src/hotspot/cpu/aarch64/globals_aarch64.hpp

```diff
@@ -33,9 +33,15 @@
 // (see globals.hpp)
 
 define_pd_global(bool, ImplicitNullChecks,       true);  // Generate code for implicit null checks
-define_pd_global(bool, TrapBasedNullChecks,  false);
+define_pd_global(bool, TrapBasedNullChecks,     false);
 define_pd_global(bool, UncommonNullCast,         true);  // Uncommon-trap NULLs past to check cast
 
+#if COMPILER2_OR_JVMCI
```

Comment:
I find it easier to read when only the default value is guarded. In that respect, `COMPILER2_OR_JVMCI` code diverges from the rest of the code base.

Reply:
So you want something similar to `COMPILER1_AND_COMPILER2_PRESENT()` used for `CodeCacheSegmentSize` in following line?
Yes, we have similar macro for JVMCI and C2. How about this:
```
define_pd_global(bool, MoveIntrinsicStubsGen,   false COMPILER2_OR_JVMCI_PRESENT( || true));
```

Reply:
What about the following?
```
define_pd_global(bool, MoveIntrinsicStubsGen,  NOT_COMPILER2_OR_JVMCI(false) COMPILER2_OR_JVMCI_PRESENT(true));
```

Or even `NOT_COMPILER2_OR_JVMCI` maybe?
```
define_pd_global(bool, MoveIntrinsicStubsGen, COMPILER2_OR_JVMCI);
```
Are you concerned it is too long?

Reply:
Yes, I don't want long line. I will try the last suggestion. I was able to build locally with it on linux.